src/interfaces/token.spec.ts

```diff
@@ -787,19 +794,53 @@ describe("TokenInterface", () => {
           sendTransaction: expect.any(Function),
         }
       );
-      expect(approveMock).toHaveBeenCalledTimes(1);
-      expect(approveMock).toHaveBeenCalledWith(spenderAddress, amount, {});
+      expect(populateApproveMock).toHaveBeenCalledTimes(1);
+      expect(populateApproveMock).toHaveBeenCalledWith(spenderAddress, amount, {});
+    });
+
+    it("should throw when approving native token", async () => {
+      try {
+        await tokenInterface.populateApprove(ownerAddress, ZeroAddress, spenderAddress, amount);
+      } catch (error) {
+        expect(error).toStrictEqual(new SdkError(`Native tokens cant be approved: ${ZeroAddress}`));
+        expect(Contract).not.toHaveBeenCalled();
+        expect(populateApproveMock).not.toHaveBeenCalled();
+      }
+    });
+
+    it("should throw if approving token as its spender", async () => {
+      try {
+        await tokenInterface.populateApprove(ownerAddress, spenderAddress, spenderAddress, amount);
+      } catch (error) {
+        expect(error).toStrictEqual(new SdkError(`Cant approve token as its spender: ${spenderAddress}`));
+        expect(Contract).not.toHaveBeenCalled();
+        expect(populateApproveMock).not.toHaveBeenCalled();
+      }
+    });
+  });
+
+  describe("approve", () => {
+    beforeEach(() => {
+      sendTransactionMock.mockResolvedValue(true);
+      tokenInterface.populateApprove = populateApproveMock;
+    });
+
+    it("should return a transaction response when approving non native token", async () => {
+      const approveResult = await tokenInterface.approve(ownerAddress, tokenAddress, spenderAddress, amount);
+
+      expect(approveResult).toEqual(true);
+      expect(populateApproveMock).toHaveBeenCalledTimes(1);
+      expect(populateApproveMock).toHaveBeenCalledWith(ownerAddress, tokenAddress, spenderAddress, amount, {});
       expect(sendTransactionMock).toHaveBeenCalledTimes(1);
-      expect(sendTransactionMock).toHaveBeenCalledWith("approved");
+      expect(sendTransactionMock).toHaveBeenCalledWith({ data: "PopulatedTransactionData" });
```

Comment:
it should not return this right? This does not return a populated transaction

Reply:
It's not a return, but a "called with"

Reply:
The `sendTransaction` is executed with the `populatedTransaction` object as parameter, that results from `populateApprove`